src/data/arena-tag.ts

```diff
@@ -1389,6 +1389,74 @@ class GrassWaterPledgeTag extends ArenaTag {
   }
 }
 
+/**
+ * Class to describe the field effect of G-Max moves that damage all Pokemon that
+ * are not immune by 1/6th of their health
+ *
+ * Used in:
+ * G-Max Vine Lash: Grass
+ * G-Max Wildfire: Fire
+ * G-Max Cannonade: Water
+ * G-Max Volcalith: Rock
+ */
+export class TypeImmuneDamageOverTimeTag extends ArenaTag {
+  private immuneType: Type;
+
+  constructor(tagType, sourceMove: Moves, sourceId: number, side: ArenaTagSide, immuneType: Type) {
+    super(tagType, 4, sourceMove, sourceId, side);
+    this.immuneType = immuneType;
+  }
+
+  private getAnimationForType() {
+    switch (this.immuneType) {
+      case Type.GRASS:
+        return CommonAnim.WRAP;
+      case Type.FIRE:
+        return CommonAnim.FIRE_SPIN;
+      case Type.WATER:
+        return CommonAnim.WHIRLPOOL;
+      case Type.ROCK:
+        return CommonAnim.SAND_TOMB;
```

Comment:
```suggestion
        return CommonAnim.SALT_CURE;
```
This animation might work better